src/controllers/healthcheck/healthcheck.controller.js

```diff
@@ -0,0 +1,192 @@
+ * tx-mining-service are optionally included in the checks, depending on the parameters.
+ *
+ * Also returns the http status code and the overall status of the health check.
+ *
+ * Returned object format:
+ *     {
+ *       httpStatus: 200,
+ *       status: 'pass',
+ *       checks: {
+ *        'Wallet health_wallet': [
+ *          {
+ *            componentName: 'Wallet health_wallet',
+ *            componentType: 'internal',
+ *            status: 'fail',
+ *            output: 'Wallet is not ready. Current state: Syncing',
+ *            time: expect.any(String),
+ *          },
+ *        ],
+ *        'Fullnode http://fakehost:8083/v1a/': [
+ *          {
+ *            componentName: 'Fullnode http://fakehost:8083/v1a/',
+ *            componentType: 'fullnode',
+ *            status: 'pass',
+ *            output: 'Fullnode is responding',
+ *            time: expect.any(String),
+ *          },
+ *        ],
+ *        'TxMiningService http://fake.txmining:8084/': [
+ *          {
+ *            componentName: 'TxMiningService http://fake.txmining:8084/',
+ *            componentType: 'service',
+ *            status: 'pass',
+ *            output: 'Tx Mining Service is healthy',
+ *            time: expect.any(String),
+ *          },
+ *        ],
+ *      }
+ *    }
+ *
+ */
+async function getWalletsHealthChecks(
+  walletIds,
+  includeFullnode = false,
+  includeTxMiningService = false
+) {
+  const promises = Array.from(walletIds).map(
+    walletId => healthService.getWalletHealth(initializedWallets.get(walletId), walletId)
+  );
+
+  if (includeFullnode) {
+    promises.push(healthService.getFullnodeHealth());
+  }
+  if (includeTxMiningService) {
+    promises.push(healthService.getTxMiningServiceHealth());
+  }
+
+  const resolvedPromises = await Promise.all(promises);
+
+  const checks = {};
+
+  for (const healthData of resolvedPromises) {
+    // We use an array as the value to stick to our current format,
+    // which allows us to add more checks to a component if needed
+    checks[healthData.componentName] = [healthData];
+  }
+
+  const httpStatus = resolvedPromises.every(healthData => healthData.status === 'pass') ? 200 : 503;
+  // If any of the checks failed, the status is fail. If all checks passed, the status is pass.
+  // Otherwise, the status is warn.
+  let status = httpStatus === 200 ? 'pass' : 'warn';
+  status = resolvedPromises.some(healthData => healthData.status === 'fail')
+    ? 'fail'
+    : status;
+
+  return { checks, httpStatus, status };
+}
+
+/**
+ * Controller for the /health endpoint that returns the health
+ * of the wallet-headless service, including all started wallets,
+ * the connected fullnode and the tx-mining-service
+ */
+async function getGlobalHealth(req, res) {
+  const sendError = message => {
+    res.status(400).send({
+      success: false,
+      message,
+    });
+  };
+
+  if (!('x-wallet-ids' in req.headers)) {
```

Comment:
Why are you using the parameters to filter in the headers and not as get parameters? 

And why creating two endpoints to get the health, when it could be only one with optional parameters to return the fullnode/txmining health?

Reply:
> Why are you using the parameters to filter in the headers and not as get parameters?

I was following the same pattern used in the wallet endpoints: https://github.com/HathorNetwork/hathor-wallet-headless/blob/master/src/middlewares/wallet.middleware.js#L25, with the difference that we accept a comma-separated list in the healthcheck.

But this is the only reason, I could get from the GET parameters as well.

> And why creating two endpoints to get the health, when it could be only one with optional parameters to return the fullnode/txmining health?

I you're talking about the `/health` vs `/health/wallet`, I guess this became a little confusing indeed after the last refactors.

I think we could remove the `/health/wallet` one.

Reply:
Done in https://github.com/HathorNetwork/hathor-wallet-headless/pull/348/commits/55e74889ecdeeabebd7f39b4f86e6eec1c89903c